### PR TITLE
feat: anon-aadhaar specs raw

### PR DESCRIPTION
**Pull request by Meyanis95**
This PR is adding the technical specifications for the Anon Aadhaar protocol.

### CODE REVIEW

**Review by oskarth — COMMENTED** [review]
I think this is a good start!

I added some comments. Some can be addressed in PR, and others can be left for later.

I suggest creating a GH issue for "Make Anon-Aadhaar spec draft status" to make it clear what needs to be done. Even if it is just tightening up spec in general (e.g. comments such as above).

Thought experiment: Imagine you are Indian government and you know nothing about ZK or Ethereum, how will AA protocol help people? what are dependencies and possible issues etc?

Another way to keep scope tight: Defer specific mechanisms to other specs. E.g. Ethereum on-chain verification. Thought experiment here: 1) Ethereum not necessary 2) You can imagine using some other "on-chain" verification mechanism

File: specs/anon-aadhaar/specs.md [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+  tags:
+- zero-knowledge
+- identity
+- privacy
+
+---
+
+# Change Process
+
+This document is governed by the [1/COSS](https://rfc.vac.dev/spec/1/).
```
[the remark this answers is not in the document]
Comment (Meyanis95) [reply specs/anon-aadhaar/specs.md:23]: Do we have an actual link to point to? Or not yet?

[the remark this answers is not in the document]
Comment (oskarth) [reply specs/anon-aadhaar/specs.md:23]: this works for now I think! https://github.com/zkspecs/zkspecs/tree/main/specs/1

File: specs/anon-aadhaar/specs.md [hunk 2]
```diff
@@ -0,0 +1,542 @@
+## Anon-Aadhaar spec raw
+
+---
+
+slug: CS-02
+title: CS-02/ANON-AADHAAR-V2
+name: Anonymous Aadhaar Verification Protocol
+status: raw
```
[the remark this answers is not in the document]
Comment (Meyanis95) [reply specs/anon-aadhaar/specs.md:8]: I will currently promote as draft, as I think the current version is reflecting what the protocol is currently doing

File: specs/anon-aadhaar/specs.md [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+```
+
+## Circuit Design
+
+### Inputs
+
+#### Private Inputs
+
+- `qrDataPadded: Bytes[]` - Padded signed data from QR code
+- `qrDataPaddedLength: BigInt` - Length of the the QR code padded data
```
Comment (dmpierre) [inline specs/anon-aadhaar/specs.md:143]: Should it explain what is meant by `BigInt`?

File: specs/anon-aadhaar/specs.md [hunk 4]
```diff
@@ -0,0 +1,542 @@
...
+2. RSA PKCS#1 v2.1
+3. SHA-256 FIPS 180-4
+
+# Appendix A: Circuit Architecture
+
+## Components
+
+### 1. AadhaarQRVerifier
+
+**Description**: Verifies the Aadhaar QR data using RSA signature.
```
Comment (dmpierre) [inline specs/anon-aadhaar/specs.md:357]: I would maybe say that further info can be found in the appendix.

**Review by dmpierre — COMMENTED** [review]
Thanks, this really clears up quite a few things for me!

File: specs/anon-aadhaar/specs.md (hunk 3 above)
Comment (oskarth) [reply specs/anon-aadhaar/specs.md:143]: Yes good point, maybe a brief reference in prelude of section or so

**Review by oskarth — APPROVED** [review]
Thanks, great! LGTM as a first solid draft. We can always iterate on it.

As we merge this, to close GH issue https://github.com/zkspecs/zkspecs/issues/3, would you mind to also create very brief follow up issue(s)? So it is clear what next steps are.

E.g. something like:

> Issue created for improvements/companion specs (e.g. POD, VC interface)

but up to you based on what you think is highest priority for AA/ZK-ID effort. It can also just be one general issue with short checklist items like `[ ] POD AA interface spec` etc.

File: specs/anon-aadhaar/specs.md [hunk 5]
```diff
@@ -0,0 +1,542 @@
...
+Circuit constraints:
+
+1. RSA signature validity
+2. Field bounds checking
+
+Note that the circuit is agnostic in terms of RSA public key, it check generate a valid proof as soon as the data format correspond to the Aadhaar Secure QR code, it's the role of the verifier to check the correctness of the used RSA public key, and check that the outputed hash of the public key, is corresponding to one official UIDAI public key.
+
+### 3. Verification
+
+Verifier MUST check:
```
[the remark this answers is not in the document]
Comment (Meyanis95) [reply specs/anon-aadhaar/specs.md:236]: I don't have a clear framework, but it shapes the protocol. The specifications mirror the protocol design and the security choices made when building it.

File: specs/anon-aadhaar/specs.md [hunk 6]
```diff
@@ -0,0 +1,542 @@
...
+4. Create zero-knowledge proof
+
+Circuit constraints:
+
+1. RSA signature validity
+2. Field bounds checking
+
+Note that the circuit is agnostic in terms of RSA public key, it check generate a valid proof as soon as the data format correspond to the Aadhaar Secure QR code, it's the role of the verifier to check the correctness of the used RSA public key, and check that the outputed hash of the public key, is corresponding to one official UIDAI public key.
+
+### 3. Verification
```
[the remark this answers is not in the document]
Comment (Meyanis95) [reply specs/anon-aadhaar/specs.md:234]: It's already the case but maybe you think of another format to present it? 
But I missed one I'll add it

File: specs/anon-aadhaar/specs.md (hunk 3 above)
Comment (Meyanis95) [reply specs/anon-aadhaar/specs.md:143]: This type of interface is linked to the proving system used, so I'm not sure about the right level of abstraction here. 
Should it just specify the format of the data to be processed, or match the exact data type used in the current reference implementation? 
What are your thoughts?

File: specs/anon-aadhaar/specs.md (hunk 4 above)
Comment (Meyanis95) [reply specs/anon-aadhaar/specs.md:357]: Yes! 
The appendix are referenced above

Comment (dmpierre) [reply specs/anon-aadhaar/specs.md:357]: Woops, sorry I think I missed my line count!

File: specs/anon-aadhaar/specs.md [hunk 7]
```diff
@@ -0,0 +1,542 @@
...
+const verified = await verify(anonAadhaarProof);
+```
+
+# References
+
+1. [UIDAI QR Code Specification](https://uidai.gov.in/images/resource/User_manulal_QR_Code_15032019.pdf)
+2. RSA PKCS#1 v2.1
+3. SHA-256 FIPS 180-4
+
+# Appendix A: Circuit Architecture
```
[the remark this answers is not in the document]
Comment (dmpierre) [reply specs/anon-aadhaar/specs.md:351]: Agree! (Related to my misplaced previous comment)

File: specs/anon-aadhaar/specs.md [hunk 8]
```diff
@@ -0,0 +1,542 @@
...
+### 3. Verification
+
+Verifier MUST check:
+
+1. Proof validity
+2. RSA public key validity
+
+Verifier SHOULD check:
+
+1. Nullifier uniqueness, by checking that the nullifier seed is the one corresponding to the action set.
```
[the remark this answers is not in the document]
Comment (Meyanis95) [reply specs/anon-aadhaar/specs.md:243]: I see! 
Let me resolve it

File: specs/anon-aadhaar/specs.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (Meyanis95) [reply specs/anon-aadhaar/specs.md:351]: Yes! 
I resolve this and added mentions to appendix earlier in the specs, but let me know if you think of other relevant places where it's missing

**Comment by oskarth** [comment]
@dmpierre once you approve feel free to merge @Meyanis95 (also see comment above)

**Pull request merged**